src/app/jobads/currentJobAds.tsx

```diff
@@ -0,0 +1,37 @@
+import ImageCard from '@/components/ImageCard/ImageCard'
+import { readJobAdsCurrentAction } from '@/actions/jobAds/read'
+
+type PropTypes = {
+    not?: number
+}
+
+/**
+ * @param not - pass it not: a id of a jobad to exclude from the list
+ */
+export default async function CurrentJobAds({ not }: PropTypes) {
+    const res = await readJobAdsCurrentAction()
+    if (!res.success) {
+        throw res.error ?
+            new Error(res.error[0].message) :
+            new Error('unknown error reading jobad')
```

Comment:
I think you should check for if the error code is unauthorized or unauthenticated. Then you could show the 404 page (or an unauthorized page if we get that in the future).

Another option would be to check if the user has the correct permissions at the beginning, but then as you have pointed out earlier we would be checking the users permissions twice. Another issue with this is that if the action permissions were to change you would also have to change the permissions in the component.

Maybe, since this is a common occurrence, we could write a monad similar to useActionCall that would check the error code beforehand and if it's unauthorized or unauthenticated it would do the appropriate action.

Reply:
We have talked about this before. If (!res.success) we should have a standard unwrapper function for actionerrors

Reply:
Forgot about that, but I agree.